website/static/js/filepage/index.js

```diff
@@ -363,6 +363,64 @@ var FileViewPage = {
         }, 1000);
 
         self.mfrIframeParent = $('#mfrIframeParent');
+        function goToRevisions(e){
+          var editable = self.editor && self.editor.selected;
+          var viewable = self.mfrIframeParent.is(':visible');
+          if (editable || viewable){
+              if (viewable){
+                  self.mfrIframeParent.toggle();
+              }
+              if (editable) {
+                  self.editor.selected = false;
+              }
+          }
+          self.revisions.selected = true;
+        }
+
+        function changeVersionHeader(value){
+          m.render(document.getElementById('version-link'), m('a', {onclick: goToRevisions},'(Version:  ' + String(value) + ')'));
+        }
+
+        function httpGetAsync(theUrl, callback) {
+            var xmlHttp = new XMLHttpRequest();
```

Comment:
I don't understand, that is the way to get a json response. Are you saying I should use JSONRequest instead? This code just issues a request and returns data that is parsed as a json. The method to get that json shouldn't matter. Unless you specifically want me to use Ajax or jQuery this method works without relying on importing any more libraries or using too much overhead.